lib/src/contextual-menu/ContextualMenu.tsx

```diff
@@ -1,63 +1,86 @@
 
-const DxcContextualMenu = ({ items, defaultSelectedItemIndex = -1 }: MenuPropsType) => {
-  const [selectedItemIndex, setSelectedItemIndex] = useState<number>(defaultSelectedItemIndex);
+export const ContextualMenuContext = createContext<ContextualMenuContextProps | null>(null);
 
-  const renderSingleItem = (item: Item, index: number) => (
-    <Li key={`option-${index}`} role="menuitem">
-      <MenuItemAction
-        {...item}
-        selected={selectedItemIndex === index}
-        onSelect={() => {
-          setSelectedItemIndex(index);
-          item.onSelect?.();
-        }}
-      />
-    </Li>
-  );
+const isGroupItem = (item: Item | GroupItem): item is GroupItem => "items" in item;
+const isSection = (item: Section | Item | GroupItem): item is Section => "items" in item && !("label" in item);
 
-  let accLength = 0;
-  const renderSection = (section: SectionType, currentSectionIndex: number, items: SectionType[]) => {
-    const startingIndex = accLength;
-    accLength += section.items.length;
-    return (
-      <Fragment key={`separator-${currentSectionIndex}`}>
-        <Li role="group">
-          {section.title != null && <Title>{section.title}</Title>}
-          <Section>{section.items.map((item, index) => renderSingleItem(item, startingIndex + index))}</Section>
-        </Li>
-        {currentSectionIndex !== items.length - 1 && <Divider aria-hidden />}
-      </Fragment>
+const addIdToItems = (items: ContextualMenuPropsType["items"]): (ItemWithId | GroupItemWithId)[] | SectionWithId[] => {
+  let accId = 0;
+  const innerAddIdToItems = (items: ContextualMenuPropsType["items"]) => {
+    return items.map((item: Item | GroupItem | Section) =>
+      isSection(item)
+        ? { ...item, items: innerAddIdToItems(item.items) }
+        : isGroupItem(item)
+        ? { ...item, items: innerAddIdToItems(item.items) }
+        : { ...item, id: accId++ }
     );
   };
+  return innerAddIdToItems(items);
+};
 
-  return (
-    <Menu role="menu">
-      {items.map((item: Item | SectionType, index: number, items: MenuPropsType["items"]) =>
-        "items" in item ? renderSection(item, index, items as SectionType[]) : renderSingleItem(item, index)
+const DxcContextualMenu = ({ items }: ContextualMenuPropsType) => {
+  const [selectedItemId, setSelectedItemId] = useState(-1);
+  const itemsWithId = useMemo(() => addIdToItems(items), [items]);
+
+  const renderSection = (section: SectionWithId, currentSectionIndex: number, length: number) => (
+    <Fragment key={`section-${currentSectionIndex}`}>
+      <li role="group">
+        {section.title != null && <Title>{section.title}</Title>}
+        <SectionList>
+          {section.items.map((item, index) => (
+            <MenuItem item={item} key={`${item.label}-${index}`} />
+          ))}
+        </SectionList>
+      </li>
+      {currentSectionIndex !== length - 1 && (
+        <DxcInset top="0.25rem" bottom="0.25rem">
+          <DxcDivider color="lightGrey" decorative />
```

Comment:
DxcDivider decorative prop is true by default

Reply:
Fixed!